radix-engine-interface/src/api/component_api.rs

```diff
@@ -25,12 +16,16 @@ pub trait ClientComponentApi<E> {
 
     fn new_key_value_store(&mut self) -> Result<KeyValueStoreId, E>;
 
-    fn globalize_component(&mut self, component_id: ComponentId) -> Result<ComponentAddress, E>;
+    fn globalize(&mut self, node_id: RENodeId) -> Result<ComponentAddress, E>;
```

Comment:
I think it depends on how we store things in the call frame (ie if we keep any type information in the ID). I've been leaning towards not keeping any type information in the ID and it being totally maintained by the System/Kernel so that this interface would just accept some `object_handle`, possibly being even only `u32` size once local handles are implemented.

Reply:
It depends on how we implement local handles _(to some extend, the `RENodeId` is already like a global handle, which is checked globally by heap and track, rather than locally by the call frame)._ 

It's a problem of how much "static" info we expose to the clients through the "handle", like whether to be able to tell if it's an object handle, or a package handle, or substate handle. 

I'm leaning towards using different handle for the 4 types our system maintains. Like in Java, there is `class` (two index bytes) vs `objectref` in JVM instruction set. Fully dynamic handle may make our interface much more harder to reason about.

Reply:
As discussed, we will re-look into this as we refactor these interfaces in the next few PRs.